app/models/file_metadata.rb

```diff
@@ -10,16 +10,23 @@ class FileMetadata < Valkyrie::Resource
   attribute :original_filename, Valkyrie::Types::Set
   attribute :file_identifiers, Valkyrie::Types::Set
   attribute :use, Valkyrie::Types::Set
+  attribute :size, Valkyrie::Types::Set
 
   def self.for(file:)
     new(label: file.original_filename, original_filename: file.original_filename, mime_type: file.content_type, use: file.try(:use) || [Valkyrie::Vocab::PCDMUse.OriginalFile])
   end
 
   def original_file?
-    use.include?(Valkyrie::Vocab::PCDMUse.OriginalFile)
+    use_uris.include?(Valkyrie::Vocab::PCDMUse.OriginalFile)
   end
 
   def derivative?
-    use.include?(Valkyrie::Vocab::PCDMUse.ServiceFile)
+    use_uris.include?(Valkyrie::Vocab::PCDMUse.ServiceFile)
   end
+
+  private
+
+    def use_uris
+      use.map { |u| u.is_a?(Hash) ? u[:@id] : u }
```

Comment:
Why's this necessary? Valkyrie should support mapping back/forth to RDF::URI.

Reply:
When the use values are coming back from Solr, they come back as hashes like `{"@id":"http://pcdm.org/use#OriginalFile"}` — I'll try to figure out why Valkyrie isn't converting them to URIs.

Reply:
Oh, it's probably because the use values are nested objects inside the FileMetadata nested objects.